middleware/auth.js

```diff
@@ -0,0 +1,61 @@
+export default async function (context) {
+  if (!context.from) {
+    if (context.route.query.code) {
+      context.app.$cookies.set('auth-token', context.route.query.code, {
+        secure: true,
+        sameSite: 'Strict',
+        httpOnly: true,
+      })
+
+      return context.redirect(context.route.path)
+    }
+
+    if (context.app.$cookies.get('auth-token')) {
+      const cookie = context.app.$cookies.get('auth-token')
+
+      await context.store.dispatch('auth/fetchUser', { token: cookie })
+    }
+  }
+
+  if (routeOption(context.route, 'auth', false)) return
```

Comment:
What are these supposed to check? They aren’t self documenting enough, some commentary is required

Reply:
This is taken from: https://github.com/nuxt-community/auth-module/blob/dev/src/core/middleware.ts#L6